idaes/power_generation/carbon_capture/mea_solvent_system/unit_models/column.py

```diff
@@ -485,66 +472,268 @@ def _make_performance(self):
+                                    self.vapor_phase.length_domain,
+                                    rule=rule_flood_velocity,
+                                    doc='Gas velocity at flooding point')
+        else:
+            # Stichlmair flooding correlation
+            # Dimensionless pressure drop term (deltaP/rho_l*g*Z) at flooding
+            self.deltaP_dimless_term = Var(self.flowsheet().time,
+                                    self.liquid_phase.length_domain,
+                                    units=None,
+                                    initialize=0.001,
+                                    doc='Dimensionless pressure drop term at flooding')
+            
+            # Flooding point correlation
+            def rule_flood_eqn(blk, t, x):
+                Fr_l = ((blk.velocity_liq[t, x])**2)*blk.a_ref/(g_ref*(blk.eps_ref)**4.65)
+                h0 = 0.555*(Fr_l)**(1/3)
+                return (blk.deltaP_dimless_term[t, x])**(-2) - ((40*(2+blk.c[1-x])*h0/3)/\
+                    (1 - blk.eps_ref + h0*(1 + 20*(blk.deltaP_dimless_term[t, x])**2))) - \
+                    (186*h0/(blk.eps_ref - h0*(1 + 20*(blk.deltaP_dimless_term[t, x])**2))) == 0
+            
+            self.flooding_equation = Constraint(self.flowsheet().time,
+                                    self.liquid_phase.length_domain,
+                                    rule=rule_flood_eqn,
+                                    doc='Flooding point correlation')
+            
+            # Gas velocity at flooding point
+            def rule_flood_velocity(blk, t, x):
+                return ((blk.deltaP_dimless_term[t, 1-x]*\
+                        blk.liquid_phase.properties[t, 1-x].mw/\
+                        blk.liquid_phase.properties[t, 1-x].vol_mol_phase['Liq']*g_ref)/\
+                    ((3/4)*blk.f0_wet[t, 1-x]*((1-blk.holdup_vap[t, x])/\
+                        (blk.holdup_vap[t, x])**4.65)*blk.vapor_phase.properties[t, x].mw/\
+                        blk.vapor_phase.properties[t, x].vol_mol_phase['Vap']*\
+                     /blk.dp_wet[t, 1-x]))**0.5
+                    
+            self.gas_velocity_flood = Expression(self.flowsheet().time,
+                                    self.vapor_phase.length_domain,
+                                    rule=rule_flood_velocity,
+                                    doc='Gas velocity at flooding point')
+            
+        # Flooding fraction 
+        def rule_flood_fraction(blk, t, x):
+            return blk.velocity_vap[t, x]/blk.gas_velocity_flood[t, x]
+            
+        self.flood_fraction = Expression(self.flowsheet().time,
+                                    self.vapor_phase.length_domain,
+                                    rule=rule_flood_fraction,
+                                    doc='Flooding fraction (expected to be below 0.8)')
+        
+        # Mass and heat transfer terms
+        
+        # Mass transfer terms
+        self.pressure_equil = Var(
```

Comment:
For the future, we probably want to think about this some more. In simpler models (without the film theory), this comes directly from the property packages.